go-controller/pkg/cni/cniserver_linux.go

```diff
@@ -0,0 +1,56 @@
+// +build linux
+
+package cni
+
+import (
+	"fmt"
+	"net"
+	"os"
+	"path/filepath"
+
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	utilwait "k8s.io/apimachinery/pkg/util/wait"
+)
+
+// Start the Server's local HTTP server on a root-owned Unix domain socket.
+// requestFunc will be called to handle pod setup/teardown operations on each
+// request to the Server's HTTP server, and should return a PodResult
+// when the operation has completed.
+func (s *Server) Start(requestFunc cniRequestFunc) error {
+	if requestFunc == nil {
+		return fmt.Errorf("no pod request handler")
+	}
+	s.requestFunc = requestFunc
+
+	// Remove and re-create the socket directory with root-only permissions
+	if err := os.RemoveAll(s.rundir); err != nil && !os.IsNotExist(err) {
+		utilruntime.HandleError(fmt.Errorf("failed to remove old pod info socket: %v", err))
+	}
+	if err := os.RemoveAll(s.rundir); err != nil && !os.IsNotExist(err) {
```

Comment:
We are running the same command twice above. What am I missing here?

Reply:
You're not missing anything.

Reply:
Fixed.